ui/pages/smart-transactions/smart-transaction-status-page/smart-transaction-status-page.tsx

```diff
@@ -0,0 +1,314 @@
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          alignItems={AlignItems.center}
+        >
+          <Box display={Display.Flex} style={{ fontSize: '48px' }}>
+            <Icon
+              name={iconName}
+              color={iconColor}
+              size={IconSize.Inherit}
+              marginBottom={4}
+            />
+          </Box>
+          <Text
+            color={TextColor.textDefault}
+            variant={TextVariant.headingMd}
+            as="h4"
+            fontWeight={FontWeight.Bold}
+          >
+            {title}
+          </Text>
+        </Box>
+        {isSmartTransactionPending && (
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Column}
+            alignItems={AlignItems.center}
+            width={BlockSize.Full}
+          >
+            <div className="smart-transaction-status-page__loading-bar-container">
+              <div
+                className="smart-transaction-status-page__loading-bar"
+                style={{
+                  width: `${
+                    (100 / stxDeadline) *
+                    (stxDeadline - timeLeftForPendingStxInSec)
+                  }%`,
+                }}
+              />
+            </div>
+          </Box>
+        )}
+        {description && (
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Column}
+            alignItems={AlignItems.center}
+          >
+            <Text
+              marginTop={2}
+              color={TextColor.textAlternative}
+              variant={TextVariant.bodySm}
+            >
+              {description}
+            </Text>
+          </Box>
+        )}
+        {blockExplorerUrl && (
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Column}
+            marginTop={2}
+          >
+            <Button
+              type="link"
+              variant={ButtonVariant.Link}
+              onClick={() => {
+                global.platform.openTab({ url: blockExplorerUrl });
+                if (!isSmartTransactionPending) {
+                  onCloseExtension();
+                }
+              }}
+            >
+              {t('viewTransaction')}
+            </Button>
+          </Box>
+        )}
+      </Box>
+      <Box
+        className="smart-transaction-status-page__footer"
+        display={Display.Flex}
+        flexDirection={FlexDirection.Column}
+        width={BlockSize.Full}
+        padding={4}
+        paddingBottom={0}
+      >
+        {isDapp && !isSmartTransactionPending && (
+          <ButtonSecondary
+            data-testid="smart-transaction-status-page-footer-close-button"
+            onClick={onCloseExtension}
+            width={BlockSize.Full}
+            marginTop={3}
+          >
+            {t('closeExtension')}
+          </ButtonSecondary>
+        )}
+        {!isDapp && (
+          <ButtonSecondary
```

Comment:
For a large JSX block like this, I'm a huge fan of many simple local functions to act as mini-components.

Could we break this up for easier maintenance into functions such as `Deadline`, `Description`, `BlockExplorerLink`, `CloseExtensionButton` and `ViewActivityButton`?